sdk/scheduler/src/main/java/org/apache/mesos/scheduler/plan/Step.java

```diff
@@ -26,17 +26,25 @@
      * @see {@link #updateOfferStatus(Collection<Offer.Operation>)} which returns the outcome of the
      *      {@link OfferRequirement}
      */
-    Optional<OfferRequirement> start();
+    Optional<PodInstanceRequirement> start();
 
     /**
      * Notifies the Step whether the {@link OfferRequirement} previously returned by
-     * {@link #start()} has been successfully accepted/fulfilled. The {@code operations} param is
+     * {@link #getOfferRequirement()} has been successfully accepted/fulfilled. The {@code operations} param is
      * empty when no offers matching the requirement previously returned by {@link #clone()
-     * could be found. This is only called if {@link #start()} returned a non-{@code null}
+     * could be found. This is only called if {@link #getOfferRequirement()} returned a non-{@code null}
      * {@link OfferRequirement}.
      */
     void updateOfferStatus(Collection<Offer.Operation> operations);
 
+    Optional<String> getAsset();
+
+    /**
+     * Provides the assets which the Step is currently operating on to avoid contention.
+     * @return
```

Comment:
- `assets` -> `asset`
- remove stub `@return`
- could add `, or an empty Optional if no asset is currently being operated on`
- maybe describe what the string is supposed to be the name of? or give an example? from further down it looks like these should be pod names? maybe just call them dirtyPods?